tools/c7n_azure/c7n_azure/actions/logic_app.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from azure.mgmt.logic import logic_management_client
+
+from c7n.actions.webhook import Webhook
+from c7n.utils import type_schema
+
+
+class LogicAppAction(Webhook):
+    """Calls an Azure Logic App with optional parameters
+       and body populated from JMESPath queries.
+
+        .. code-block:: yaml
+
+          policies:
+            - name: call-logic-app
+              resource: azure.vm
+              description: |
+                Call logic app with list of VM's
+              actions:
+               - type: logic-app
+                 resource-group: custodian-test
+                 logic-app-name: cclogicapp
+                 batch: true
+                 body: 'resources[].{ vm_name: name }'
+    """
+
+    schema = type_schema(
```

Comment:
if a filter or action is associated to more than one resource, you want to set `schema_alias = True` as a class level var that will automatically generate a global definition and ref in the associated resources schema during jsonschema gen, and it will be used by the ref doc system to note common filters/patterns for a provider.

Reply:
in this case its inheriting from webhook, which i modified in a recent to include this which this class will inherit, so this is fyi

Reply:
Yeah I saw you changed that and I wasn't sure what it did, so thank you for the info. :)